Activate all hosting features modal: Improve design

allilevine opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/7727

## Proposed Changes

* Update modal copy and styles to match [the design](https://github.com/Automattic/dotcom-forge/issues/7727#issuecomment-2340253121).
* The new styles will be applied to other modals: warnings when adding themes or plugins.

Before | After
--- | ----
<img width="778" alt="Screen Shot 2024-09-16 at 3 54 54 PM" src="https://github.com/user-attachments/assets/90f79633-676a-4dd8-9fda-896a8a4f82bd"> | <img width="732" alt="Screen Shot 2024-09-17 at 4 07 46 PM" src="https://github.com/user-attachments/assets/9547fc2a-742d-4942-800d-24b8e30a13f2">
<img width="277" alt="Screen Shot 2024-09-17 at 10 26 44 AM" src="https://github.com/user-attachments/assets/1eecd915-860a-40ae-acd1-c05f004da344"> | <img width="282" alt="Screen Shot 2024-09-17 at 4 08 03 PM" src="https://github.com/user-attachments/assets/5a048fa1-2632-4370-8bcc-6f43c7772377">
<img width="273" alt="Screen Shot 2024-09-17 at 10 26 51 AM" src="https://github.com/user-attachments/assets/bc00ed02-ca18-41ad-9a7c-0698ba692fea"> | <img width="282" alt="Screen Shot 2024-09-17 at 4 08 22 PM" src="https://github.com/user-attachments/assets/3bea5358-34e4-44f2-8fa1-7feee5d2d498">


Additional planned changes for separate PRs:
- Bold the text "your domain will change."
- Change the data center picker to match the design.
- Update the close button to use the WordPress icon.
- Add snackbar notification.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Improve the modal design and keep this interaction exciting for the user.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to /hosting-features for a site that has a Business plan, but isn't yet Atomic.
* Click "Activate now."
* View the modal at different screen sizes.
* Go to /plugins and click to add a plugin. Check the modal at different screen sizes.
* Go to /themes and click to install a theme. Check the modal at different screen sizes.
* Go to the Design Picker page with a Marketplace theme selected /setup/update-design/designSetup?siteSlug=:site&theme=:theme_slug. Theme slugs such as makoney and olsen-fse can be used. Check the modal at different screen sizes.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

allilevine committed:
Update copy and styles for eligibility warnings.

allilevine committed:
Adjust font sizes and weights.

allilevine committed:
Style mobile padding and CTA.

kristastevens commented:
Hi @allilevine, thanks for the copy ping! This looks fine.

allilevine commented:
> Hi @allilevine, thanks for the copy ping! This looks fine.

@kristastevens Thank you for taking a look!

allilevine committed:
Remove translation check.

allilevine committed:
Style domain cards for mobile.

allilevine committed:
Prevent widows.

allilevine commented on client/blocks/eligibility-warnings/index.tsx:
```diff
@@ -98,7 +99,8 @@ export const EligibilityWarnings = ( {
 			'eligibility-warnings--blocking-hold': hasBlockingHold( listHolds ),
-			'eligibility-warnings--without-title': context !== 'plugin-details',
+			'eligibility-warnings--without-title':
+				context !== 'plugin-details' && context !== 'hosting-features',
```
I'm removing this class for hosting features, which I've added a title for.

allilevine commented on client/blocks/eligibility-warnings/index.tsx:
```diff
@@ -197,6 +199,16 @@ export const EligibilityWarnings = ( {
 				</CompactCard>
 			) }
 
+			{ context === 'hosting-features' && (
```
This is the hosting features title.

allilevine commented on client/blocks/eligibility-warnings/index.tsx:
```diff
@@ -303,6 +315,9 @@ function getProceedButtonText(
 	if ( siteRequiresGoingPublic( holds ) ) {
 		return translate( 'Make your site public and continue' );
 	}
+	if ( context === 'hosting-features' ) {
```
This is the hosting features CTA text.

allilevine commented on client/blocks/eligibility-warnings/style.scss:
```diff
@@ -1,19 +1,25 @@
 	// 16px from card padding + 32 = 48 gap from top|bottom
 	// 24px from card padding + 24 = 48 gab from left|right
-	padding: 32px 24px;
+	padding: 24px;
```
Changing the padding to be 48px all around (modal padding + eligibility warnings padding).

allilevine commented on client/blocks/eligibility-warnings/style.scss:
```diff
@@ -1,19 +1,25 @@
 	box-shadow: 0 0 0 1px var(--color-border-subtle);
 
+	@media (max-width: 660px) {
+		padding: 8px;
```
Reducing the padding on mobile.

allilevine commented on client/blocks/eligibility-warnings/style.scss:
```diff
@@ -1,19 +1,25 @@
 
 		@media (max-width: 660px) {
-			display: block;
+			flex-direction: column-reverse;
```
Moving the support block under the CTA on mobile.

allilevine commented on client/blocks/eligibility-warnings/style.scss:
```diff
@@ -220,14 +238,37 @@
+		}
+	}
+
+	@media (max-width: 660px) {
```
Move the badges above the URLs on mobile. I could also use flexbox on the desktop view, but I wasn't sure if there's a reason that it's positioned absolutely.

allilevine commented on client/blocks/eligibility-warnings/warning-list.tsx:
```diff
@@ -31,7 +35,7 @@ export const WarningList = ( { context, translate, warnings, showContact = true
 				<div className="eligibility-warnings__warning" key={ index }>
 					<div className="eligibility-warnings__message">
-						{ context !== 'plugin-details' && (
+						{ context !== 'plugin-details' && context !== 'hosting-features' && (
```
I'm hiding the message name returned from the backend, for hosting features, to match the design.

allilevine commented on client/blocks/eligibility-warnings/warning-list.tsx:
```diff
@@ -121,25 +124,16 @@ function getWarningDescription(
-							args: warningCount,
-						}
-				  );
+			return preventWidows(
```
I removed the `hasTranslation` check because the i18n check says that all strings are translated. I also added widow prevention.

allilevine commented on client/blocks/eligibility-warnings/warning-list.tsx:
```diff
@@ -15,14 +15,18 @@ interface ExternalProps {
+						<span
+							className={ `eligibility-warnings__message-description ${
+								context === 'hosting-features' &&
+								'eligibility-warnings__message-description--hosting-features'
```
I added a custom class to change the font weight here. Maybe unnecessary if we're okay with the font weight changing elsewhere.

allilevine commented:
@javierarce It would be great if you could take a look! I've deferred a couple of changes from the design to follow-up PRs:

- Bold the text "your domain will change."
- Change the data center picker to match the design.
- Add snackbar notification.

javierarce commented:
> @javierarce It would be great if you could take a look! I've deferred a couple of changes from the design to follow-up PRs

@allilevine sure, I'll review it!

javierarce commented:
First of all, thanks for the fixes, it looks much nicer already!

Here's what I saw in mobile:

- The padding around the modal is still too big (`37px`). It would be great to reduce it to `32px`.
- The gap between the badge and the URL is `10px` instead of `4px`.
- There's also extra padding between the URL box at the bottom and the gray area.

In desktop:

- The padding is too big: `72px` instead of `48px`.

And some general observations:

- Could we reduce the separation between the URL boxes to `4px` instead of `5px`? I've already updated that in the designs.
- Could we round the gray box with a `border-radius` of `4px`? The data center box will also be rounded, and it look nicer if both use the same radius. 
- Could we replace the close icon with the version from the WordPress Design Library?
- Could we remove the border around the modal? This was my fault because I used a border to make it look nicer when I shared the designs, but we don't really need it.

javierarce commented:
> Could we remove the border around the modal? This was my fault because I used a border to make it look nicer when I shared the designs, but we don't really need it.

@allilevine my bad, other modals have a border 🤦‍♂️ Let's not change that for now.

allilevine committed:
Adjust gaps and padding.

allilevine committed:
Remove bottom padding and margin from last child eligibility warning.

allilevine commented:
@javierarce Thank you for the feedback!

> * The padding around the modal is still too big (`37px`). It would be great to reduce it to `32px`.
> * The padding is too big: `72px` instead of `48px`.

Thanks for catching this. I added up the modal and warning padding, but I missed that each warning card has additional padding. I've updated the desktop padding to add up to 48px and the mobile padding to add up to 32px.

> * The gap between the badge and the URL is `10px` instead of `4px`.

Fixed on mobile.

> * There's also extra padding between the URL box at the bottom and the gray area.

Fixed everywhere by setting the last child padding to 0.

> * Could we reduce the separation between the URL boxes to `4px` instead of `5px`? I've already updated that in the designs.

Reduced the bottom margin to 4px.

> * Could we round the gray box with a `border-radius` of `4px`? The data center box will also be rounded, and it look nicer if both use the same radius.

I'm seeing a `border-radius` of `4px` when I look at the CSS and the computed value. It's this gray box, right?
<img width="732" alt="Screen Shot 2024-09-17 at 4 07 46 PM" src="https://github.com/user-attachments/assets/e6ff50e7-66fa-40df-b5bd-a716a63e659f">

> * Could we replace the close icon with the version from the WordPress Design Library?

Do you mind if I make that replacement in a separate PR? It's touching a different component. Also, is it the close icon from @wordpress/icons?

javierarce commented:
Thanks, it looks much better now!

> I'm seeing a border-radius of 4px when I look at the CSS and the computed value. It's this gray box, right?

Yes, that's it. I probably missed that 🙈 

> Do you mind if I make that replacement in a separate PR? It's touching a different component. Also, is it the close icon from @wordpress/icons?

Of course! And yes, it's that icon.

javierarce reviewed (approved):
🚀

allilevine commented:
I noticed other issues with the close button at <375px wide screens that I'll address separately:

<img width="289" alt="Screen Shot 2024-09-18 at 9 23 54 AM" src="https://github.com/user-attachments/assets/5f170e87-ae92-40d4-8cec-beacca104eeb">

<img width="207" alt="Screen Shot 2024-09-18 at 9 31 28 AM" src="https://github.com/user-attachments/assets/4446f7e0-8d91-4629-8db0-008696860463">

Addison-Stavlo commented on client/blocks/eligibility-warnings/warning-list.tsx:
```diff
@@ -1,9 +1,9 @@
 import { Fragment } from 'react';
 import ActionPanelLink from 'calypso/components/action-panel/link';
 import ExternalLink from 'calypso/components/external-link';
+import { preventWidows } from 'calypso/lib/formatting';
```
I didn't realize we had/used this formatting function. I think we generally handle them with css now, but support is varied so this seems fine.

Addison-Stavlo commented on client/blocks/eligibility-warnings/style.scss:
```diff
@@ -166,6 +180,10 @@
 		color: var(--color-text-subtle);
 		font-size: $font-body;
+
+		&--hosting-features {
```
TIL!

allilevine committed:
Remove widow prevention, not needed.

Addison-Stavlo commented on client/blocks/eligibility-warnings/warning-list.tsx:
```diff
@@ -121,25 +123,14 @@ function getWarningDescription(
-							args: warningCount,
-						}
-				  );
+			return translate(
+				'By proceeding the following change will be made to the site:',
+				'By proceeding the following changes will be made to the site:',
```
I see we are both getting rid of the old `hasTranslation` check while adding a new translation. I see the `stringFreeze` tag is here so Im assuming the plan is to hold for main translations?

allilevine replied:
@Addison-Stavlo The new copy is somehow already translated: https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/hosting-features-activate-modal

Maybe it was used before?

<img width="722" alt="Screen Shot 2024-09-18 at 11 35 23 AM" src="https://github.com/user-attachments/assets/fdd68b78-614b-4fa6-a422-d19423632878">

Addison-Stavlo replied:
awesome!

DustyReagan reviewed (approved):
Looks good to me! I tested using your instructions and with some old instructions for [excessive disk space warning](https://github.com/Automattic/wp-calypso/pull/64388), and [activating backups](https://github.com/Automattic/wp-calypso/pull/68421) looking for style issues. I didn't find any regressions.

Addison-Stavlo reviewed (approved):
Looks great!
